go/tasks/plugins/array/k8s/launcher.go

```diff
@@ -94,6 +96,16 @@ func LaunchSubTasks(ctx context.Context, tCtx core.TaskExecutionContext, kubeCli
 
 		pod = ApplyPodPolicies(ctx, config, pod)
 
+		// Allocate Token
+		resourceNamespace := core.ResourceNamespace(pod.Namespace)
+		allocationStatus, err := tCtx.ResourceManager().AllocateResource(ctx, resourceNamespace, pod.Name, core.ResourceConstraintsSpec{})
```

Comment:
Nob question, here we allocate resources using pod.Namespace, by on the initialization step, we :
```
iCtx.ResourceRegistrar().RegisterResourceQuota(ctx, core.ResourceNamespace(primaryLabel), tokenLimit)
```

Does that mean that we only can request tokens with the same label we used during registration? If so, does enforcement of limits per namespace would require registering quotas for each namespace on initialization?
@kumare3 ?

Reply:
https://github.com/lyft/flyteplugins/blob/master/go/tasks/plugins/hive/executor.go#L127?

Reply:
I also have the same question about how resource namespace and requests are used

Reply:
I don't think I understand your question, but let me provide some details regarding RM to see if that resolves your concern.
 
RM is basically a pooling system for plugins. A plugin requests to create one or more pools during setup time (using `ResourceRegistrar().RegisterResourceQuota()`). At the end of the setup time, the RM will be created based on the valid pool-creation requests, and the actual pools will be created. During execution time, the plugin send token-allocate and token-release requests to RM. Note that the term`ResourceNamespace` here has nothing to do with the users' namespace or k8s namespace (i.e. `<project>-<domain>`). `ResourceNamespace` here just means the name of a token pool, nothing else. So how you want to use it is totally up to you. 

I just realized that the argument names used in the function signature are misleading in this case. Those names made sense when I first wrote RM, but it seems like it has become confusing for its current form. I'll try to refactor the naming soon. Sorry for the confusion.